### PR TITLE
Update taint-mode docs

### PR DESCRIPTION
**Pull request by IagoAbal**
Follows returntocorp/semgrep#4901

**NOTE:** This should be merged _after_ release 0.87.0 is out.

# Thanks for improving Semgrep Docs 😀

### Please ensure

- [x] A subject matter expert (SME) reviews the content
- [x] A technical writer reviews the content or PR
- [x] This change has no security implications or else you have pinged the security team
- [x] Redirects are added if the PR changes page URLs

### CODE REVIEW

**Comment by adamkvitek** [comment]
Hi there @IagoAbal, thanks for the input, made a tiny quick review here, please check (just made the last part's sentence flow a little better): https://github.com/returntocorp/semgrep-docs/pull/488/commits/bd8110942ad5db9fb75b224c7924337c185cec3d

(Not blocking comment.) Not sure if we should mention release notes - we do not have them currently (do you want to have this merge by the release 0.87.0 or earlier, even now?). On the other hand, we can take the info from the release notes and just put it there later, so this may serve as a sort of TODO for me.

**Comment by IagoAbal** [comment]
I didn't want to repeat the reasons for the change there in the docs. But this PR is not meant to be merged until next week after 0.87.0 is out. I clarified this is in the PR comment. 👍

**Comment by adamkvitek** [comment]
> I didn't want to repeat the reasons for the change there in the docs. But this PR is not meant to be merged until next week after 0.87.0 is out. I clarified this is in the PR comment. 👍

Thank you! I see the changelog entry, will work on it on Monday! Looks great and informative.

**Review by adamkvitek — APPROVED** [review]

File: docs/writing-rules/data-flow/taint-mode.md [hunk 1]
```diff
@@ -132,17 +132,19 @@ pattern-sanitizers:
...
+Semgrep can treat metavariables with the same name as the _same_ metavariable. This can be done through rule `options` by setting `taint_unify_mvars: true`. Unification enforces that whatever a metavariable binds to in each of these operators is, syntactically speaking, the **same** piece of code. For example, if a metavariable binds to a code variable `x` in the source match, it must bind to the same code variable `x` in the sink match. In general, unless you know what you are doing, avoid metavariable unification between sources and sinks.
 
-<iframe src="https://semgrep.dev/embed/editor?snippet=Q64P" border="0" frameBorder="0" width="100%" height="435"></iframe>
+:::info
+Semgrep used to have a different behavior, see release notes for 0.87.0 for more information.
+:::
 
-For it to work, you must use different metavariables:
+For example, here is a rule where unification of source and sink metavariables is useful:
```
Comment (adamkvitek) [inline docs/writing-rules/data-flow/taint-mode.md:148]:
```suggestion
The following example demonstrates the use of source and sink metavariables.```

**Comment by adamkvitek** [comment]
@IagoAbal are we going to merge this?

File: docs/writing-rules/data-flow/taint-mode.md [hunk 2]
```diff
@@ -132,17 +132,19 @@ pattern-sanitizers:
...
+The patterns specified by `pattern-sources` and `pattern-sinks` (and `pattern-sanitizers`) are all independent of each other. If a metavariable used in `pattern-sources` has the same name as a metavariable used in `pattern-sinks`, these are still different metavariables.
 
-However, you typically do not want to use the same metavariable across these different operators, since it may have unintended consequences! Unless you have a specific goal in mind, you should probably use different metavariables when matching sources, sanitizers, and sinks.
+In the message of a taint-mode rule, you can refer to any metavariable bound by `pattern-sinks`, as well as to any metavariable bound by `pattern-sources` that does not conflict with a metavariable bound by `pattern-sinks`.
```
[the remark this answers is not in the document]
Comment (IagoAbal) [reply docs/writing-rules/data-flow/taint-mode.md:140]: The rule message is the `message` field of the rule.

File: docs/writing-rules/data-flow/taint-mode.md (hunk 1 above)
Comment (IagoAbal) [reply docs/writing-rules/data-flow/taint-mode.md:148]: We're not demonstrating metavariables, we're demonstrating the unification of metavariables.

**Comment by IagoAbal** [comment]
I don't know what is wrong with the license/cla check 🤷‍♂️

**Comment by adamkvitek** [comment]
@IagoAbal Will try to check the error message if there will be one... let's see. Thanks for working on this!

File: docs/writing-rules/data-flow/taint-mode.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (adamkvitek) [reply docs/writing-rules/data-flow/taint-mode.md:140]: Maybe we can specify it this way...

**Comment by adamkvitek** [comment]
@pabloest please, could you help us to check this out? Not sure what is the issue with license/cla.

**Pull request merged**